crates/bitcoin-client/src/actor/client.rs

```diff
@@ -88,7 +99,39 @@ impl Handler<GetBlockMessage> for BitcoinClientActor {
         _ctx: &mut ActorContext,
     ) -> Result<bitcoin::Block> {
         let GetBlockMessage { hash } = msg;
-        Ok(self.retry(|| self.rpc_client.get_block(&hash)).await?)
+
+        let rpc_ret = self.retry(|| self.rpc_client.get_block(&hash)).await;
+        if let Ok(block) = rpc_ret {
+            return Ok(block);
+        }
```

Comment:
Maybe read the local store first, then get from rpc? the local store can be a cache.

Reply:
local block store will only helpful when:

1. for online server, may need reorg block in startup for executor catch up sequencer
2. for offline execution, we need to upload these reorg to public storage, users could downloads these block

For normal process, this cache cannot help in any way

Reply:
I mean, the local store saves all the blocks.